samples/drivers/counter/alarm/src/main.c

```diff
@@ -74,7 +78,11 @@ static void test_counter_interrupt_fn(const struct device *counter_dev,
 
 void main(void)
 {
-	const struct device *const counter_dev = DEVICE_DT_GET(TIMER);
+#if defined(CONFIG_COUNTER_GECKO_STIMER)
+	const struct device *counter_dev = DEVICE_DT_GET(STIMER);
+#else
+	const struct device *counter_dev = DEVICE_DT_GET(TIMER);
+#endif
```

Comment:
why is this needed?

Reply:
This was initially placed there because of warnings about naming conflicts caused by `TIMER(n)` macro defined in SiLabs HAL. This macro is actually not used so I decided to rename it (https://github.com/zephyrproject-rtos/hal_silabs/pull/19/commits/81c3c5b6b8a57a8be1ce22b58c2f9c7e76096936). Now we don't have to use the `#if` you pointed out and I was able to remove it.